internal/verifier/change_stream.go

```diff
@@ -52,8 +53,81 @@ func (uee UnknownEventError) Error() string {
 	return fmt.Sprintf("received event with unknown optype: %+v", uee.Event)
 }
 
+type ChangeStreamReader struct {
+	readerType clusterType
+
+	lastChangeEventTime *primitive.Timestamp
+	logger              *logger.Logger
+	namespaces          []string
+	metaDBName          string
+
+	metaClient    *mongo.Client
```

Comment:
A change stream “reader” shouldn’t ordinarily need to access the metadata. Can this be separated out?

If we do need to keep this here, this might as well be `metaDB *mongo.Database` and combine metaDBName and metaClient.